mne/tests/test_dipole.py

```diff
@@ -150,4 +156,59 @@ def test_len_index_dipoles():
     _compare_dipoles(d_mask, dipole[idx])
 
 
+@requires_sklearn
+@testing.requires_testing_data
+def test_min_distance_fit_dipole():
+
+    data_path = testing.data_path()
+    raw_fname = data_path + '/MEG/sample/sample_audvis_trunc_raw.fif'
+
+    subjects_dir = op.join(data_path, 'subjects')
+    fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis-cov.fif')
+    fname_trans = op.join(data_path, 'MEG', 'sample',
+                          'sample_audvis_trunc-trans.fif')
+    fname_bem = op.join(subjects_dir, 'sample', 'bem',
+                        'sample-1280-1280-1280-bem-sol.fif')
+
+    subject = 'sample'
+
+    raw = Raw(raw_fname, preload=True)
+
+    # select eeg data
+    picks = pick_types(raw.info, meg=False, eeg=True, exclude='bads')
+    info = pick_info(raw.info, picks)
+
+    # Let's use cov = Identity
+    cov = read_cov(fname_cov)
+    cov['data'] = np.eye(cov['data'].shape[0])
+
+    # Simulated scal map
+    simulated_scalp_map = np.zeros(picks.shape[0])
+    simulated_scalp_map[27:34] = 1
+
+    simulated_scalp_map = simulated_scalp_map[:, None]
+
+    evoked = EvokedArray(simulated_scalp_map, info, tmin=0)
+
+    min_dist = 5.  # distance in mm
+
+    dip, residual = fit_dipole(evoked, cov, fname_bem, fname_trans,
+                               min_dist=min_dist)
+
+    dist = _compute_depth(dip, fname_bem, fname_trans, subject, subjects_dir)
+
+    assert_true(min_dist < (dist[0] * 1000.) < (min_dist + 1.))
+
+
+def _compute_depth(dip, fname_bem, fname_trans, subject, subjects_dir):
+    trans = read_trans(fname_trans)
+    trans = _get_mri_head_t(trans)[0]
+    bem = read_bem_solution(fname_bem)
+    surf = _bem_find_surface(bem, 'inner_skull')
```

Comment:
@jmontoyam @Eric89GXL I had to use the inner_skull surface from bem solution to get the proper dist value.